### DISCUSSION TITLE
Fetching Most Popular Courses in Sunbird

### DISCUSSION THREAD

**Discussion by komalm** [Sunbird-Lern/Community#200] [Query]
Hello @AmiableAnil @vinukumar-vs,

We are currently working with **Sunbird version 4.10.0** and are required to fetch the first 10 most popular courses. We are considering the following criteria to determine popularity:

1. Courses with the maximum number of enrollments.
2. Most viewed courses.
3. Most issued certificate courses.

However, we are unsure about how to retrieve this data. Could you please guide us on how to fetch this information using the Sunbird API or any other method available on the platform?

We appreciate your assistance on this matter.

cc @paritshivani

**Comment by komalm** [comment]
Hello Team,

I hope this message finds you well.

I am following up on our previous request regarding the **retrieval of the top 10 most popular courses** in Sunbird version 4.10.0. We are still awaiting guidance on how to fetch this data based on maximum enrollments and most views.
Your prompt attention to this matter would be greatly appreciated as it is crucial for our ongoing work.

Thank you for your time and assistance.

**Reply by AmiableAnil** [reply]
@komalm There are no existing API is available for this. But the data is available in cassandra [sunbird_courses.user_enrolments](https://lern.sunbird.org/use/developer-guide/lms-service/data-models/cassandra/data-dictionary#sunbird_courses.user_enrolments-primary-key-userid-courseid-batchid). From where you can the data for **Courses with the maximum number of enrollments** and **Most issued certificate courses**. For the **Most viewed courses**, START telemetry is triggered from the portal or mobile, so you can use that data to get this information.
So you need to writer a wrapper to fulfil your requirement.
